dubbo-common/src/main/java/org/apache/dubbo/config/AbstractInterfaceConfig.java

```diff
@@ -252,7 +252,7 @@
         }
         if (CollectionUtils.isNotEmpty(this.registries)) {
             this.registries.forEach(registryConfig -> {
-                if (registryConfig.getScopeModel() != applicationModel) {
+                if (registryConfig != null && registryConfig.getScopeModel() != applicationModel) {
```

Comment:
**Scalability Issue**: Adding a null-check before accessing the 'getScopeModel()' method can prevent potential NullPointerExceptions, which can affect the scalability and robustness of the system by ensuring that operations on collections of registry configurations are safe even when some elements might be null. <br> **Fix**: Implement null-checks when working with collections to prevent NullPointerExceptions, enhancing the system's ability to scale by ensuring stability across operations. <br> **Code Suggestion**: 
 ```
 -                if (registryConfig.getScopeModel() != applicationModel) {
 +                if (registryConfig != null && registryConfig.getScopeModel() != applicationModel) {
 ```